lib/aggregations/class-post-date.php

```diff
@@ -39,15 +42,84 @@ public function __construct( DSL $dsl, array $args ) {
 		parent::__construct( $dsl, $args );
 	}
 
+	/**
+	 * Get the date range for a queried date based on the interval.
+	 * Since a post cannot be published at multiple DateTimes, only one (the first)
+	 * queried post_date value is applied to the DSL.
+	 *
+	 * @param string $queried_date The queried date value.
+	 *
+	 * @return array An array containing timestamps for from and to. Null if no query is present.
+	 */
+	private function get_date_range( string $queried_date ) : array {
+		switch ( $this->interval ) {
+			case 'year':
+				// Since we want all the months in a single year, set interval to months and offset by eleven months.
+				$interval = 'M';
+				$offset   = 11;
+				// TODO ensure that the offset date is 01-01 00:00:00, rather than 01-31 23:59:59.
```

Comment:
Given this, I am considering ditching a DRY method for determining date ranges and just going with hard coded ranges for each interval.